### PR TITLE
Use 'opt-level = 1' for all non libparsec_* dependencies in dev

### PR DESCRIPTION
**Pull request by touilleMan**
# What has changed ?

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -37,11 +37,13 @@ tokio = { version = "1.25", features = ["rt", "rt-multi-thread"] }
...
-# Rust unoptimized code is so slow that we customize dev profile to enable some
-# optimizations, otherwise Python tests take forever.
+# Rust unoptimized code is really slow with crypto algorithm and regex compilation,
+# hence we optimize our 3rd party dependencies.
+[profile.dev.package."*"]
+opt-level = 1
+
 [profile.dev-python]
 inherits = "dev"
-opt-level = 1
```
Comment (touilleMan) [inline Cargo.toml:44]: not sure if we should remove this: the difference is without this the libparsec_* crates won't be optimized
@FirelightFlagboy or @TimeEngineer can you do a small bench to see the difference in compilation time & running the python tests

File: Cargo.toml [hunk 2]
```diff
@@ -37,11 +37,13 @@ tokio = { version = "1.25", features = ["rt", "rt-multi-thread"] }
 lazy_static = "1.4.0"
 futures = "0.3.21"
 
-# Rust unoptimized code is so slow that we customize dev profile to enable some
-# optimizations, otherwise Python tests take forever.
+# Rust unoptimized code is really slow with crypto algorithm and regex compilation,
+# hence we optimize our 3rd party dependencies.
+[profile.dev.package."*"]
+opt-level = 1
```
Comment (TimeEngineer) [inline Cargo.toml:43]: why `opt-1`, not `opt-2` ?

Comment (FirelightFlagboy) [reply Cargo.toml:43]: The difference between `opt-1` and `opt-2` was 20s of increase compilation time for `opt-2` for a gain of `200ms` for executing the test (on @touilleMan computer)

File: Cargo.toml (hunk 1 above)
Comment (FirelightFlagboy) [reply Cargo.toml:44]: ## Compilation time

### With your current change

```shell
$ cargo clean
$ python make.py r
[...]
    Finished dev-python [unoptimized + debuginfo] target(s) in 1m 14s
```

### With `profile.dev-python.opt-level = 1`

```shell
$ sed '/profile.dev-python/a opt-level = 1' Cargo.toml
$ cargo clean
$ python make.py r
[...]
    Finished dev-python [optimized + debuginfo] target(s) in 2m 01s
```

## Test time

### With your current change

```shell
$ pytest -n 4 tests
[...]
1208 passed, 386 skipped, 1 xfailed in 14.23s
```

### With `profile.dev-python.opt-level = 1`

```shell
$ pytest -n 4
[...]
1208 passed, 386 skipped, 1 xfailed in 11.95s
```

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**